### PR TITLE
test: add e2e tests

### PR DESCRIPTION
**Pull request by SgtPooki**
- **test: add e2e tests**
- **chore: generate fixtures script can start/stop kubo daemon for you**
- **chore: generate-fixtures-script allows paths**
- **chore: rename generate fixtures script**
- **test: migrate test fixture/helper**
- **test: add test for bug #461**

backport for e2e test to prevent regression of https://github.com/ipfs/ipld-explorer-components/issues/461

Also note that this started from the e2e test config from `ipfs-webui` and the existing [`explore.test.js` tests](https://github.com/ipfs/ipfs-webui/blob/502abd44586ebd878ce298f48de614b45cb32139/test/e2e/explore.test.js) there

### CODE REVIEW

File: test/e2e/setup/ipfs-backend.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  // const { apiHost, apiPort } = { apiHost: '127.0.0.1', apiPort: rpcPort }
+
+  if (String(apiPort) !== rpcPort) {
+    console.error(`Invalid RPC port returned by IPFS backend: ${apiPort} != ${rpcPort}`)
+    await ipfsd.stop()
+    process.exit(1)
+  }
+
+  // const rpcAddr = `/ip4/${apiHost}/tcp/${apiPort}`
+  // const gatewayAddr = endpoint ?? rpcAddr
```
Comment (SgtPooki) [inline test/e2e/setup/ipfs-backend.ts:71]:
```suggestion
```

File: test/e2e/setup/ipfs-backend.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    ipfs = ipfsd.api
+  }
+  const { id, agentVersion } = await ipfs.id()
+  const { gateway, api: rpcAddr } = await ipfsd.info()
+  const rpcApiMaddr = multiaddr(rpcAddr)
+  const { address: apiHost, port: apiPort } = rpcApiMaddr.nodeAddress()
+  const { hostname: gatewayHost, port: gatewayPort } = new URL(gateway)
+
+  // some temporary hardcoding until https://github.com/ipfs/js-ipfsd-ctl/issues/831 is resolved.
+  // const { apiHost, apiPort } = { apiHost: '127.0.0.1', apiPort: rpcPort }
```
Comment (SgtPooki) [inline test/e2e/setup/ipfs-backend.ts:62]:
```suggestion
```

File: test/e2e/setup/ipfs-backend.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  if (ipfsd != null && ipfs != null) {
+    throw new Error('IPFS backend already running')
+  }
+  const endpoint = process.env.E2E_API_URL
+  if (endpoint != null && endpoint !== '') {
+    // create http rpc client for endpoint passed via E2E_API_URL=
+    ipfs = create(endpoint)
+  } else {
+    // use ipfds-ctl to spawn daemon to expose http api used for e2e tests
+    // gatewayPort = await getPort(gatewayPort, '0.0.0.0')
```
Comment (SgtPooki) [inline test/e2e/setup/ipfs-backend.ts:28]:
```suggestion
```

File: test/e2e/setup/ipfs-backend.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
+// @ts-check
```
Comment (SgtPooki) [inline test/e2e/setup/ipfs-backend.ts:1]:
```suggestion
```

File: test/e2e/setup/global-setup.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  // Read and expose backend info in env availables inside of test() blocks
+  const { rpcAddr, id, agentVersion, apiOpts, kuboGateway } = JSON.parse(fs.readFileSync(path.join(__dirname, 'ipfs-backend.json'), 'utf8'))
+  process.env.IPFS_RPC_ADDR = rpcAddr
+  process.env.IPFS_RPC_ID = id
+  process.env.IPFS_RPC_VERSION = agentVersion
+
+  await ensureKuboDaemon(apiOpts)
+
+  // Set and save RPC API endpoint in storageState, so test start against
+  // desired endpoint
```
Comment (SgtPooki) [inline test/e2e/setup/global-setup.ts:51]:
```suggestion
```

File: test/e2e/setup/global-setup.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+const __filename = fileURLToPath(import.meta.url)
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const __dirname = path.dirname(__filename)
+
+// make sure that ipfs-backend is fully running
+interface ApiOpts {
+  protocol: string
+  host: string
+  port: number
+}
```
Comment (SgtPooki) [inline test/e2e/setup/global-setup.ts:16]:
```suggestion
interface ApiOpts {
  protocol: string
  host: string
  port: number
}
// make sure that ipfs-backend is fully running
```

File: test/e2e/setup/ipfs-backend.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const __filename = fileURLToPath(import.meta.url)
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const __dirname = path.dirname(__filename)
+const { console } = globalThis
+
+let ipfsd: KuboNode
+let ipfs: KuboRPCClient
+async function run (rpcPort?: string): Promise<void> {
+  // let gatewayPort = 8080
```
Comment (SgtPooki) [inline test/e2e/setup/ipfs-backend.ts:18]:
```suggestion
```

File: test/e2e/edge-cases.test.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import { test } from '@playwright/test'
+import { create, type KuboRPCClient } from 'kubo-rpc-client'
+import { loadBlockFixtures } from './fixtures/load-block-fixtures.js'
+import { testExploredCid } from './fixtures/test-explore-cid.js'
+
+test.describe('edge-cases', () => {
+  /**
+   * @type {ReturnType<import('kubo-rpc-client')['create']>}
+   */
```
Comment (SgtPooki) [inline test/e2e/edge-cases.test.ts:9]:
```suggestion
```
